lib/smart_answer/state_resolver.rb

```diff
@@ -0,0 +1,78 @@
+module SmartAnswer
+  class StateResolver
+    def initialize(flow)
+      @flow = flow
+    end
+
+    def state_from_response_store(response_store, requested_node = nil)
+      forwarding_responses = @flow.response_store == :session ? {} : response_store.all
```

Comment:
So forwarding responses (maybe could be aptly named) are previous responses that we need persist to the next node. So for query-parameter flows, we need to "re-populate" the query string with the users previous responses when we re-direct than. However, we don't for session based flows, because they just persist in the user's browser as a session cookie between requests.

Reply:
Thank you!